scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModel.scala

```diff
@@ -308,32 +311,64 @@ abstract class H2OMOJOModel
         _.getAsDouble(),
         $(defaultThreshold)))
     setOutputParameters(outputJson)
+    h2oMojoModel = mojoModel
   }
 
   private[sparkling] def setOutputParameters(outputSection: JsonObject): Unit = {}
 
-  private[sparkling] def setEasyPredictModelWrapperConfiguration(
-      config: EasyPredictModelWrapper.Config): EasyPredictModelWrapper.Config = {
-    config.setConvertUnknownCategoricalLevelsToNa(this.getConvertUnknownCategoricalLevelsToNa())
-    config.setConvertInvalidNumbersToNa(this.getConvertInvalidNumbersToNa())
-    // always let H2O produce full output, filter later if required
-    config.setUseExtendedOutput(true)
-    config
+  private[sparkling] def getEasyPredictModelWrapperConfigurationInitializers()
+      : Seq[(EasyPredictModelWrapper.Config) => EasyPredictModelWrapper.Config] = {
+    val convertUnknownCategoricalLevelsToNa = this.getConvertUnknownCategoricalLevelsToNa()
+    val convertInvalidNumbersToNa = this.getConvertInvalidNumbersToNa()
+
+    ((config: EasyPredictModelWrapper.Config) =>
+      config.setConvertUnknownCategoricalLevelsToNa(convertUnknownCategoricalLevelsToNa)) ::
+      ((config: EasyPredictModelWrapper.Config) => config.setConvertInvalidNumbersToNa(convertInvalidNumbersToNa)) ::
+      ((config: EasyPredictModelWrapper.Config) => config.setUseExtendedOutput(true)) ::
+      Nil
   }
 
   private[sparkling] def loadEasyPredictModelWrapper(): EasyPredictModelWrapper = {
     val config = new EasyPredictModelWrapper.Config()
-    val mojo = H2OMOJOCache.getMojoBackend(uid, getMojo, this)
+    val mojo = unwrapMojoModel()
     config.setModel(mojo)
-    setEasyPredictModelWrapperConfiguration(config)
+    getEasyPredictModelWrapperConfigurationInitializers().foreach(_(config))
     new EasyPredictModelWrapper(config)
   }
 
   override def copy(extra: ParamMap): H2OMOJOModel = defaultCopy(extra)
+
+  val nonSerializableField = if (System.getProperty("spark.testing", "false") == "true") {
+    new NonSerializableClass()
+  } else {
+    null
+  }
+}
+
+class NonSerializableClass extends Serializable {
```

Comment:
looks a bit confusing, is that piece "test only"?

Reply:
Yes, it's only for testing purposes. I want to make sure that H2OMOJOModel is not serialized during scoring and thus it doesn't play role of a closure for scoring udfs.

Reply:
I'm open any suggestions for improvemnts.